runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -9499,16 +9500,8 @@ J9::Z::TreeEvaluator::VMmonentEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    generateS390LabelInstruction(cg, TR::InstOpCode::label, node, cFlowRegionEnd, conditions);
 
    cg->stopUsingRegister(monitorReg);
-   if (wasteReg)
-      cg->stopUsingRegister(wasteReg);
-   if (objectClassReg)
-      cg->stopUsingRegister(objectClassReg);
-   if (lookupOffsetReg)
-      cg->stopUsingRegister(lookupOffsetReg);
-   if (tempRegister && (tempRegister != objectClassReg))
+   if (tempRegister)
```

Comment:
not sure why we had `&& (tempRegister != objectClassReg)` in the first place. I couldn't find any obvious code path the connect those two registers!